plugins/dialog/guest-js/index.ts

```diff
@@ -202,6 +297,32 @@ async function save(options: SaveDialogOptions = {}): Promise<string | null> {
   return await invoke('plugin:dialog|save', { options })
 }
 
+/**
+ * The result of a message dialog.
+ *
+ * The result is a string if the dialog has custom buttons,
+ * otherwise it is one of the default buttons.
+ *
+ * @since 2.3.0
+ */
+export type MessageDialogResult = 'Yes' | 'No' | 'Ok' | 'Cancel' | (string & {})
+
+type MessageDialogResultRust =
+  | 'Yes'
+  | 'No'
+  | 'Ok'
+  | 'Cancel'
+  | { Custom: string }
```

Comment:
If we're going to convert it in the js side, what about we just do a `#[serde(untagged)]` in the rust side?